src/Tmds.Ssh/ThrowHelper.cs

```diff
@@ -33,6 +33,12 @@ public static void ThrowProtocolUnexpectedMessageId(MessageId messageId)
             throw new ProtocolException($"Unexpected message id: {messageId}.");
         }
 
+        [DoesNotReturn]
+        public static void ThrowProtocolUnexpectedSftpPacketType(SftpPacketType packetType)
+        {
+            throw new ProtocolException($"Unexpected SFTP packet type: {packetType}.");
+        }
+
```

Comment:
Can you add a separate method in `ThrowProtocolUnexpectedSftpPacketType` that does the `new ProtocolException($"Unexpected SFTP packet type: {packetType}.");`
And call that for `CreateExceptionForUnexpectedType`, and `ThrowHelper.ThrowProtocolUnexpectedSftpPacketType`.

Reply:
You mean making a function that does the throwing inside the Throw.. function and calling that?

Reply:
```cs
        [DoesNotReturn]
        public static void ThrowProtocolUnexpectedSftpPacketType(SftpPacketType packetType)
        {
            throw new CreateNewProtocolUnexpectedSftpPacketType(packetType);
        }

        public static Exception CreateNewProtocolUnexpectedSftpPacketType(SftpPacketType packetType)
        {
            return new ProtocolException($"Unexpected SFTP packet type: {packetType}.");
        }
```
and call `CreateNewProtocolUnexpectedSftpPacketType` from `CreateExceptionForUnexpectedType`.

Reply:
Actually, we can remove `ThrowProtocolUnexpectedSftpPacketType`. It's no longer being used.